src/OpenColorIO/OCIOZArchive.cpp

```diff
@@ -0,0 +1,672 @@
+{
+    return getFileStringFromArchiveFile(filepath, archivePath, &getFileBufferByPath);
+}
+
+// See header file for information.
+std::vector<uint8_t> getFileBufferFromArchiveByExtension(const std::string & extension, const std::string & archivePath)
+{
+    return getFileStringFromArchiveFile(extension, archivePath, &getFileBufferByExtension);
+}
+
+// See header file for information.
+void getEntriesMappingFromArchiveFile(const std::string & archivePath, 
+                                      std::map<std::string, std::string> & map)
+{
+    mz_zip_file *file_info = NULL;
+    int32_t err = MZ_OK;
+    void *reader = NULL;
+
+    // Create the reader object.
+    mz_zip_reader_create(&reader);
+
+    MinizipNgHandlerGuard extracterGuard(reader, false, false);
+
+    // Open the zip from file.
+    err = mz_zip_reader_open_file(reader, archivePath.c_str());
+    if (err != MZ_OK) 
+    {
+        std::ostringstream os;
+        os << "Could not open " << archivePath.c_str() << " in order to get the entries.";
+        throw Exception(os.str().c_str());
+    }
+    else
+    {
+        // Seek to the first entry in the OCIOZ archive.
+        if (mz_zip_reader_goto_first_entry(reader) == MZ_OK)
+        {
+            do
+            {
+                // Get the current entry information.
+                if (mz_zip_reader_entry_get_info(reader, &file_info) == MZ_OK)
+                {
+                    // file_info->filename is the complete path of the file from the root of the
+                    // archive.
+                    map.insert(
+                        std::pair<std::string, std::string>(
+                            file_info->filename, 
+                            std::string(file_info->filename) + std::to_string(file_info->crc)
+                        )
+                    );
+                }
+            } while (mz_zip_reader_goto_next_entry(reader) == MZ_OK);  
+        }
+    }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+//////////////////////////////////////////////////////////////////////////////////////
+// Implementation of CIOPOciozArchive class.
+//////////////////////////////////////////////////////////////////////////////////////
+
+std::vector<uint8_t> CIOPOciozArchive::getLutData(const char * filepath) const
+{
+    // In order to ease the implementation and to facilitate a future Python binding, this method
+    // uses std::vector<uint8_t> buffer instead of a std::istream.
+    //
+    // It is expected that in most cases, the compiler will be able to avoid copying the buffer
+    // (RVO/NRVO).
+    // 
+    // During testings with ocioperf, the first iteration was a little slower using the buffer
+    // instead of a std::istream (max 5%). But the following iterations are just as fast due to
+    // the FileTransform cache.
+
+    std::ifstream ociozStream = Platform::CreateInputFileStream(
```

Comment:
What is the use of `ociozStream` in this function (same question for `getConfigData()` below) ?

Reply:
Good catch.  It is an artifact from earlier implementation that didn't get refactored. I'll fix it.